src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -1456,6 +1586,14 @@ int32_t ZooKeeper::getConnectionXid() const
     return impl->getConnectionXid();
 }
 
+String ZooKeeper::getConnectedHostAvailabilityZone() const
+{
+    auto idx = impl->getConnectedNodeIdx();
+    if (idx < 0)
```

Comment:
Could not came up with the case when it could happen. May be throw LOGAL_ERROR if it some kind of impossible case?

Reply:
I thought that it can happen with TestKeeper, but it returns 0

Reply:
This is possible when session is expired